### PR TITLE
UCI - Release 4.2.0 Changes

### PR DESCRIPTION
**Pull request by ChakshuGautam**
### Changes made

- LBs added for ODK and Fusionauth since there was no way to view the UI for the same internally.
- Updated configs for UCI env.
       -  ODK Env Changes
             - Change `DB_Schema` - since `aggregate` was not automatically generated when provisioning.
             - Typo in word `postgresql` for `DB_URL`
       - UCI API env changes
             - Add private variable `ENCRYPTION_KEY`
- Update liveness and readiness probe configs for third party open sources service  - ODK, Fusionauth, and Hasura.
- Remove Private ingress configs for `odk` and `fusionauth` services.
- Add UCI service to private ingress.

#Note:  Please squash and merge. Please ignore Individual commits as those are not useful in this case and were mostly used since testing locally was not possible.

### CODE REVIEW

File: ansible/roles/stack-sunbird/templates/uci_api.env [hunk 1]
```diff
@@ -31,3 +31,5 @@ UCI_CORE_BASE_URL=http://inbound-service.{{namespace}}.svc.cluster.local:8085
 #FusionAuth
 FUSIONAUTH_KEY={{fusionauth_service_admin_key}}
```
Comment (rjshrjndrn) [inline ansible/roles/stack-sunbird/templates/uci_api.env:32]: Make sure that you've added enough comment for the variable, if the name is not self explanatory. For instance, here, what's the admin key? Is there any simple way to create the key ? `openssl rand -hex 20` ? what should be the format?
Refer: https://github.com/project-sunbird/sunbird-devops/blob/0218e3a7708eb797f8537ba143f4cc1ddd295156/private_repo/ansible/inventory/dev/Core/secrets.yml#L69

File: ansible/roles/stack-sunbird/templates/uci_api.env [hunk 2]
```diff
@@ -31,3 +31,5 @@ UCI_CORE_BASE_URL=http://inbound-service.{{namespace}}.svc.cluster.local:8085
...
 FUSIONAUTH_URL=http://fusionauth-service.{{namespace}}.svc.cluster.local:9011
+
+ENCRYPTION_KEY= {{uci_encryption_key_base64}}
```
Comment (rjshrjndrn) [inline ansible/roles/stack-sunbird/templates/uci_api.env]: Same comment as above.

**Review by rjshrjndrn — CHANGES_REQUESTED** [review]

File: ansible/roles/stack-sunbird/templates/odk.env [hunk 3]
```diff
@@ -1,7 +1,7 @@
-DB_URL=jdbc:postgres://{{uci_postgres_host}}:5432/{{uci_odk_postgres_database}}?user={{uci_postgres_user}}&password={{uci_postgres_password}}&sslmode=require
-DB_RESOURCE_NAME=jdbc/{{uci_odk_postgres_database}}
+DB_URL=jdbc:postgresql://{{uci_postgres_host}}:5432/{{uci_odk_postgres_database}}?user={{uci_postgres_user}}&password={{uci_postgres_password}}&sslmode=require
```
[the remark this answers is not in the document]
Comment (ChakshuGautam) [reply ansible/roles/stack-sunbird/templates/odk.env:1]: No. Removing this.

File: ansible/roles/stack-sunbird/templates/odk.env [hunk 4]
```diff
@@ -1,7 +1,7 @@
...
 DB_USERNAME={{uci_postgres_user}}
 DB_PASSWORD={{uci_postgres_password}}
-DB_SCHEMA=aggregate
+DB_SCHEMA=public
 DB_NAME={{uci_odk_postgres_database}}
-DB_PORT=5432
\ No newline at end of file
+DB_PORT=5432
+AGGREGATE_HOST=
```
[the remark this answers is not in the document]
Comment (ChakshuGautam) [reply ansible/roles/stack-sunbird/templates/odk.env:7]: `AGGREGATE_HOST` requires [FQDN](https://github.com/getodk/aggregate/blob/master/docs/build-and-run-a-docker-image.md#environmental-variables) to configure UI based on the reverse proxy setup that we have. I tried adding `http://odk-service.{{namespace}}.svc.cluster.local:8080` but didn't work. Need more details on how it verifies it based on [GWT](http://www.gwtproject.org/)

**Comment by ChakshuGautam** [comment]
@rjshrjndrn Updated please check.

**Review by rjshrjndrn — APPROVED** [review]

**Pull request merged**